### PR TITLE
update layui-src verson to v2.6.8

### PR DESCRIPTION
**Pull request by javabitar**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by javabitar** [comment]
this PR  can be merged, thank you!  #54372

File: types/layui-src/Layui.d.ts [hunk 1]
```diff
@@ -0,0 +1,382 @@
+interface Layui {
+    // [index:string]:any;
```
Comment (peterblazejewicz) [inline types/layui-src/Layui.d.ts:2]:
```suggestion
```

File: types/layui-src/Layui.d.ts [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+     *  实例二：{*}可匹配全部filter<br/>&nbsp;
+     *      layui.onevent("form", "select(xx)", console.log);<br/>&nbsp;
+     *      layui.event("form","select({*})",[1,2,3]);<br/>&nbsp;
+     *  实例三：filter严格匹配<br/>&nbsp;
+     *      layui.onevent("form", "select(xx)", console.log);<br/>&nbsp;
+     *      layui.event("form","select(xx)",[1,2,3]);
+     * @param modName  模块名称，比如form，<br/>&nbsp;
+     * @param events  事件，比如：select(filter)<br/>&nbsp;
+     * @param params 回调参数，作为绑定的回调函数的参数<br/>&nbsp;
+     * @param fn 不建议用
```
Comment (peterblazejewicz) [inline types/layui-src/Layui.d.ts:191]:
```suggestion
     * @param [fn] 不建议用
```

File: types/layui-src/Layui.d.ts [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+         * @since v2.5.7
+         */
+        mobile: boolean;
+        [index: string]: any;
+    };
+    // https://www.layui.com/doc/base/infrastructure.html
+    /**
+     * 对象（Array、Object、DOM 对象等）遍历，可用于取代 for 语句
+     * @param arry  Array对象
+     * @param fn 回调函数
```
Comment (peterblazejewicz) [inline types/layui-src/Layui.d.ts:165]:
```suggestion
     * @param [fn] 回调函数
```

File: types/layui-src/Layui.d.ts [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+     * 向控制台打印一些异常信息，目前只返回了 error方法：	<br/>&nbsp;
+     *   实例：layui.hint().error('出错啦');layui.hint().error('出错啦','warn');
+     */
+    hint(): { error: (msg: any, type?: string | 'log' | 'info' | 'error' | 'warn' | 'debug') => void };
+
+    // https://www.layui.com/doc/base/infrastructure.html#other
+    /**
+     * 图片预加载
+     * @param url  图片地址直接作为Image.src值
+     * @param callback  回调函数，会透出img对象
```
Comment (peterblazejewicz) [inline types/layui-src/Layui.d.ts:228]:
```suggestion
     * @param [callback]  回调函数，会透出img对象
```

File: types/layui-src/Layui.d.ts [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+     * @param callback  css文件加载成功后的回调，错误则不会调用
+     * @param cssname 用于id标识，比如xx则生成link标签的id是 layuicss-xx	<br/>&nbsp;
+     *                 如果不传，则用文件名拼接，比如layui.js则id是layuicss-layuijs
+     */
+    link(href: string, callback: () => void, cssname?: string): any;
+
+    /**
+     * 动态加载 CSS
+     * @param href  css地址
+     * @param cssname  用于id标识，比如xx则生成link标签的id是 layuicss-xx	<br/>&nbsp;
```
Comment (peterblazejewicz) [inline types/layui-src/Layui.d.ts:246]:
```suggestion
     * @param [cssname]  用于id标识，比如xx则生成link标签的id是 layuicss-xx	<br/>&nbsp;
```

File: types/layui-src/Layui.d.ts [hunk 6]
```diff
@@ -0,0 +1,382 @@
...
+     * @param modName  模块名,比如 layer,table,form等
+     * @param events 事件名
+     * @param callback  回调函数
+     */
+    onevent(modName: string, events: string, callback: (obj: any) => any): (...arg: any) => any;
+
+    // https://www.layui.com/doc/base/infrastructure.html
+    /**
+     *  获得 location.hash 路由结构，一般在单页面应用中发挥作用。
+     * @param hash  默认 location.hash
```
Comment (peterblazejewicz) [inline types/layui-src/Layui.d.ts:295]:
```suggestion
     * @param [hash]  默认 location.hash
```

File: types/layui-src/Layui.d.ts [hunk 7]
```diff
@@ -0,0 +1,382 @@
...
+     */
+    router(hash?: string): Layui.UrlHash;
+
+    // https://www.layui.com/doc/base/infrastructure.html
+    /**
+     *  本地存储 基于sessionStorage	<br/>&nbsp;
+     *   layui对 localStorage 和 sessionStorage 友好封装，可更方便地管理本地数据。
+     * @param tableName  key键， 为sessionStorage中的一个key
+     * @param row 存储的json对象数据
```
Comment (peterblazejewicz) [inline types/layui-src/Layui.d.ts:304]:
```suggestion
     * @param [row] 存储的json对象数据
```

File: types/layui-src/Layui.d.ts [hunk 8]
```diff
@@ -0,0 +1,382 @@
...
+     */
+    sessionData(tableName: string, row?: { key: string; value?: any; remove?: boolean } | null): any;
+
+    // https://www.layui.com/doc/base/infrastructure.html
+    /**
+     *  将数组中的对象按某个成员重新对该数组排序，如：	<br/>&nbsp;
+     *  layui.sort([{a: 3},{a: 1},{a: 5}], 'a') 默认升序
+     * @param obj  有比较key的对象数组
+     * @param key 对象的一个属性，用于取值比较
+     * @param desc true则启用降序， 默认false即升序
```
Comment (peterblazejewicz) [inline types/layui-src/Layui.d.ts:314]:
```suggestion
     * @param [desc] true则启用降序， 默认false即升序
```

File: types/layui-src/Layui.d.ts [hunk 9]
```diff
@@ -0,0 +1,382 @@
...
+     */
+    sort(obj: any[], key: string, desc?: boolean): any;
+
+    // https://www.layui.com/doc/base/infrastructure.html
+    /**
+     * 阻止事件冒泡
+     * @param event  阻止的事件
```
Comment (peterblazejewicz) [inline types/layui-src/Layui.d.ts:321]:
```suggestion
     * @param [event]  阻止的事件
```

File: types/layui-src/Layui.d.ts [hunk 10]
```diff
@@ -0,0 +1,382 @@
...
+     */
+    stope(event?: Event): void;
+    // https://www.layui.com/doc/base/infrastructure.html
+    /**
+     *  用于将一段 URL 链接中的 pathname、search、hash 属性值进行对象化处理	<br/>&nbsp;
+     *  参数： href 可选。若不传，则自动读取当前页面的 url（即：location.href）	<br/>&nbsp;
+     *  示例：let url = layui.url();
+     * @param href  http(s)地址
```
Comment (peterblazejewicz) [inline types/layui-src/Layui.d.ts:329]:
```suggestion
     * @param [href]  http(s)地址
```

File: types/layui-src/Layui.d.ts [hunk 11]
```diff
@@ -0,0 +1,382 @@
...
+    use<K extends keyof Layui.LayuiModuleMap>(
+        mods: K,
+        callback: (this: Layui, module: Layui.LayuiModuleMap[K]) => any,
+        exports?: any[],
+    ): { v: string };
+    /**
+     * 使用自定模块
+     * @param mods  自定义模块，非内置模块
+     * @param callback  回调函数
+     * @param exports  数组，存储对 mods解析后加载的模块
```
Comment (peterblazejewicz) [inline types/layui-src/Layui.d.ts:355]:
```suggestion
     * @param [exports]  数组，存储对 mods解析后加载的模块
```

File: types/layui-src/Layui.d.ts [hunk 12]
```diff
@@ -0,0 +1,382 @@
...
+     */
+    use(mods: string, callback: (this: Layui, module: any) => any, exports?: any[]): { v: string };
+
+    /**
+     * 使用多模块
+     * @param mods  内置或自定模块名 (若模块不存在则抛js错误，callback不会执行)
+     * @param callback  回调函数 	<br/>&nbsp;
+     *  1、不建议callback中设置参数因为没有TS约束，可用layui.xx调用具体模块;	<br/>&nbsp;
+     *  2、手动在callback回调中指定类型比如 util:layui.Util
+     * @param exports
```
Comment (peterblazejewicz) [inline types/layui-src/Layui.d.ts:365]:
```suggestion
     * @param [exports]
```

File: types/layui-src/Layui.d.ts [hunk 13]
```diff
@@ -0,0 +1,382 @@
...
+     */
+    use(mods: string[], callback: (this: Layui, ...module: any) => any, exports?: any[]): { v: string };
+
+    /**
+     * 使用特定模块
+     * @param callback  回调函数， 从 layui 2.6 开始，首个参数是callback函数，则表示引用所有内置模块到layui.xx<br/>&nbsp;
+     * @param exports  无任何用途，可不传
```
Comment (peterblazejewicz) [inline types/layui-src/Layui.d.ts:372]:
```suggestion
     * @param [exports]  无任何用途，可不传
```

File: types/layui-src/Layui.d.ts [hunk 14]
```diff
@@ -0,0 +1,382 @@
...
+     */
+    use(callback: (this: Layui, module: { config: object; time: number }) => any, exports?: any[]): { v: string };
+
+    /**
+     * 代码高亮
+     * https://www.layui.com/doc/modules/code.html
+     * @param option
```
Comment (peterblazejewicz) [inline types/layui-src/Layui.d.ts:379]:
```suggestion
     * @param [option]
```

File: types/layui-src/index.d.ts [hunk 15]
```diff
@@ -1,745 +1,18 @@
-// Type definitions for layui-src 2.5
-// Project: https://github.com/sentsin/layui
+// Type definitions for layui-src 2.6
+// Project: https://github.com/javabitar/layui
```
Comment (peterblazejewicz) [inline types/layui-src/index.d.ts:2]: Published package still points to original github project:
https://www.npmjs.com/package/layui
```bash
npm repo layui
```
why this change?

File: types/layui-src/tsconfig.json [hunk 16]
```diff
@@ -12,5 +12,5 @@
         "noEmit": true,
         "forceConsistentCasingInFileNames": true
     },
-    "files": ["index.d.ts", "layui-src-tests.ts"]
+    "files": ["index.d.ts" , "layui-src-tests.ts"]
```
Comment (peterblazejewicz) [inline types/layui-src/tsconfig.json:15]:
```suggestion
    "files": ["index.d.ts", "layui-src-tests.ts"]
```

File: types/layui-src/index.d.ts [hunk 17]
```diff
@@ -1,745 +1,18 @@
...
+
+declare global {
+    interface Window {
+        layui: Layui;
+        lay: Layui.Lay;
+        layer: Layui.Layer;
+    }
+}
+
+export = layui;
```
Comment (peterblazejewicz) [inline types/layui-src/index.d.ts:18]:
```suggestion
```
Why this change to support CommnJS (or UMD-like) modules?
I could be wrong, but I think `layyi-src` is just a global JS script, expected to be bundled in any supported way and there is no ESM or CommonJS usage/definitions in native package.
It can still be used in CJS that way I believe, without this export:
```ts
/// <reference types="layui-src" />
import "jquery";

$.when($.ready).then(() => {
  layui.use(["layer", "form"], () => {
    const layer = layui.layer,
      form = layui.form;
    layer.msg("Hello World");
  });
});
```

**Comment by javabitar** [comment]
Thank you.you are right,`layui-src` is a global JS script,there is no ESM or CommonJS usage. I had deleted the export.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by javabitar** [comment]
Ready to merge.
